### PR TITLE
[chassis] Fix test_default_route.py for chassis

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

Fix 1 issues described in https://github.com/sonic-net/sonic-mgmt/issues/7824

pass `device_neigh_metadata` in `get_uplink_ns`

Summary:
Fixes # (issue) https://github.com/sonic-net/sonic-mgmt/issues/7824

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### How did you verify/test it?
After:
```
route/test_default_route.py::test_default_route_set_src[masic-1-3] PASSED                                                                                                                                                                     [ 33%]
route/test_default_route.py::test_default_ipv6_route_next_hop_global_address[masic-1-0] PASSED                                                                                                                                                [ 66%]
route/test_default_route.py::test_default_route_with_bgp_flap[masic-1] PASSED                                                                                                                                                                 [100%]

------------------------------------------------------------------------------------------------ generated xml file: /var/src/sonic-mgmt-int/tests/logs/tr.xml -------------------------------------------------------------------------------------------------
================================================================================================================== 3 passed in 101.33 seconds ==================================================================================================================
```

### CODE REVIEW

File: tests/route/test_default_route.py [hunk 1]
```diff
@@ -107,13 +106,14 @@ def verify_default_route_in_app_db(duthost, tbinfo, af, uplink_ns, device_neigh_
...
-def test_default_route_set_src(duthosts, tbinfo):
+def test_default_route_set_src(duthosts, enum_rand_one_per_hwsku_frontend_hostname, 
+                               enum_rand_one_asic_index, tbinfo):
     """
     check if ipv4 and ipv6 default src address match Loopback0 address
 
     """
-    duthost = find_duthost_on_role(duthosts, get_upstream_neigh_type(tbinfo['topo']['type']) , tbinfo)
-    asichost = duthost.asic_instance(0 if duthost.is_multi_asic else None)
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
```
Comment (abdosi) [inline tests/route/test_default_route.py:115]: why this change ? It works fine

Comment (wenyiz2021) [reply tests/route/test_default_route.py:115]: I see this error:
>       pytest_assert(role_host, "Could not find {} duthost".format(role))
E       Failed: Could not find T3 duthost

because `get_upstream_neigh_type(tbinfo['topo']['type'])` is T3 for chassis

Comment (abdosi) [reply tests/route/test_default_route.py:115]: T2 should have T3 neighbor in the graph . Why it is failing ?

Comment (wenyiz2021) [reply tests/route/test_default_route.py:115]: tthen might be minigraph is wrong? let me confirm

File: tests/route/test_default_route.py [hunk 2]
```diff
@@ -33,15 +34,15 @@ def get_upstream_neigh(tb, device_neigh_metadata):
     if topo_cfg_facts is None:
         return upstream_neighbors
 
-    for neigh_name, neigh_cfg in list(topo_cfg_facts.items()):
+    for neigh_name, neigh_cfg in topo_cfg_facts.iteritems():
         if neigh_type not in neigh_name:
```
Comment (abdosi) [inline tests/route/test_default_route.py:38]: @wenyiz2021 why this changes ?

File: tests/route/test_default_route.py (hunk 1 above)
Comment (wenyiz2021) [reply tests/route/test_default_route.py]: @abdosi I checked Nokia nightly with this this change works, it fails on arista testbed might due to the wrong minigraph on single-asic card, please review again

File: tests/route/test_default_route.py (hunk 2 above)
Comment (wenyiz2021) [reply tests/route/test_default_route.py:38]: I copied wrong change, local branch wasn't in sync

**Comment by wenyiz2021** [comment]
@abdosi please review again, thanks!

**Review by abdosi — APPROVED** [review]
